### PR TITLE
Add support for configurable runtime full and minimal images.

### PR DESCRIPTION
**Pull request by rmr-silicom**

### CODE REVIEW

**Review by marquiz — CHANGES_REQUESTED** [review]
Thanks for the patch @rmr-silicom!

Please document the new variables in `docs/advanced/developer-guide.md´ (in the table explaining variables for build customization)

Need to sign the CLA, too 😉 

/ok-to-test

File: Makefile [hunk 1]
```diff
@@ -74,12 +77,16 @@ image: yamls
 	$(IMAGE_BUILD_CMD) --build-arg VERSION=$(VERSION) \
 	    --target full \
 	    --build-arg HOSTMOUNT_PREFIX=$(CONTAINER_HOSTMOUNT_PREFIX) \
+		--build-arg RUNTIME_IMG_FULL=$(RUNTIME_IMG_FULL) \
+		--build-arg RUNTIME_IMG_MINIMAL=$(RUNTIME_IMG_MINIMAL) \
```
[the remark this answers is not in the document]
Comment (rmr-silicom) [reply Makefile:81]: Will do.

File: Makefile [hunk 2]
```diff
@@ -37,6 +37,9 @@ IMAGE_REPO := $(IMAGE_REGISTRY)/$(IMAGE_NAME)
 IMAGE_TAG := $(IMAGE_REPO):$(IMAGE_TAG_NAME)
 IMAGE_EXTRA_TAGS := $(foreach tag,$(IMAGE_EXTRA_TAG_NAMES),$(IMAGE_REPO):$(tag))
 
+RUNTIME_IMG_FULL?=debian:buster-slim
+RUNTIME_IMG_MINIMAL?=gcr.io/distroless/base
```
[the remark this answers is not in the document]
Comment (rmr-silicom) [reply Makefile:41]: Will do

File: docs/advanced/developer-guide.md [hunk 3]
```diff
@@ -111,6 +111,8 @@ makefile overrides.
 | KUBECONFIG                 | Kubeconfig for running e2e-tests                                  | *empty*
 | E2E_TEST_CONFIG            | Parameterization file of e2e-tests (see [example][e2e-config-sample]) | *empty*
 | OPENSHIFT                  | Non-empty value enables OpenShift specific support (currently only effective in e2e tests) | *empty*
+| BASE_IMG_FULL              | Container image for target image full (--target full)             | debian:buster-slim
+| BASE_IMG_MINIMAL           | Container image for target image minimal (--target minimal)       | gcr.io/distroless/base
```
Comment (marquiz) [inline docs/advanced/developer-guide.md:115]: Suggest s/Container image/Container base image/

File: Makefile [hunk 4]
```diff
@@ -8,6 +8,8 @@ IMAGE_BUILD_CMD ?= docker build
 IMAGE_BUILD_EXTRA_OPTS ?=
 IMAGE_PUSH_CMD ?= docker push
 CONTAINER_RUN_CMD ?= docker run
+BASE_IMG_FULL ?= debian:buster-slim
+BASE_IMG_MINIMAL ?= gcr.io/distroless/base
```
Comment (marquiz) [inline Makefile:12]: nit: I think `BASE_IMAGE_FULL` and `BASE_IMAGE_MINIMAL` (without the abbreviation) would be more in line with the other variables

**Review by marquiz — APPROVED** [review]
I think this is a nice addition to the build customization palette and ready to get merged 😎 

/label tide/merge-method-squash
/lgtm

**Pull request merged**